util/s3/s3Helper.go

```diff
@@ -322,3 +322,22 @@ func GetVersionFromName(name string) (version uint64, err error) {
 	}
 	return
 }
+
+// GetVersionPartsFromVersion converts the merged version number back into parts.
+func GetVersionPartsFromVersion(version uint64) (major uint64, minor uint64, patch uint64, err error) {
+	val := version
+
+	if val < 1<<32 {
+		err = errors.New("versions below 1.0.0 not supported")
+		return
+	}
+
+	modVal := uint64(1 << 16)
+
+	patch = val % modVal
```

Comment:
nit: division (or modulo ) is a dangerous operation. it could lead to theoretical division by zero.
if you were to change it to 
```patch = val & 65536```
you'd eliminate that one.

Reply:
Good idea, but wouldn't it need to be `patch = val & 65535`? Or `val & (modVal - 1)`

And do you prefer using the actual constant or the shifted bit?

Reply:
yes. I like @algobolson proposal to use 0xffff.
If you want to merge it as is ( since it's already approved ), you can go ahead and do that.
If you'd like to use bit masks, I would be happy to approve it again.